### PR TITLE
Disable buttons if you don't have the right entitlement

### PR DESCRIPTION
**Pull request by dabbler0**
Finally finished after a bunch of design iterations. Disables navbar items that you're not allowed to access and puts up a "locked" splash screen if you manually enter those urls anyway. For pages that are still readable become write-only, the write buttons are made disabled and given a tooltip, with few exceptions. Those exceptions are in modals (where disabling "Cancel/Ok" seems confusing; they are instead removed so that the "x" button is salient) and in the Schedule Settings page where "View/Edit Details" is changed to "View Details" when read-only.

### CODE REVIEW

**Comment by wittejm** [comment]
would a video screencap be useful for showing any major design decisions to get feedback from Liz or Jason?

File: src/App.tsx [hunk 1]
```diff
@@ -39,6 +40,9 @@ function App({ history }: { history: History }) {
   const dispatch = useAppDispatch();
   const liveCalls = useLiveCalls(selected, 30000);
   const isAuthenticated = session.status === "active";
+  const canMonitorLiveCalls =
```
Comment (wittejm) [inline src/App.tsx:43]: looks like this is no longer used in this file?

File: src/pages/MessageReviewPage.tsx [hunk 2]
```diff
@@ -194,44 +202,48 @@ const MessageReviewPage = (props: Props) => {
...
-        }
-        onReject={(reason, internalComments) =>
-          flaggedMessage &&
-          rejectMessage({
-            variables: {
-              input: {
-                messageId: flaggedMessage.id,
-                statusDetails: reason,
-                internalComments,
+      {canMonitorLiveCalls ? (
```
Comment (wittejm) [inline src/pages/MessageReviewPage.tsx:205]: wasn't there the idea of splitting out message review to its own entitlement?

File: src/pages/MessageReviewPage.tsx [hunk 3]
```diff
@@ -194,44 +202,48 @@ const MessageReviewPage = (props: Props) => {
...
-        }
-        loading={rejectLoading || approveLoading}
-      />
+              update: (cache) => decrementPendingMessagesCount(cache),
+            })
+          }
+          loading={rejectLoading || approveLoading}
+        />
+      ) : (
+        <NotAllowed />
```
Comment (wittejm) [inline src/pages/MessageReviewPage.tsx:245]: same comment as above about the long ternary. I won't tag any more examples if I see them :)

File: src/pages/StaffRosterPage.tsx [hunk 4]
```diff
@@ -114,7 +114,7 @@ export default function StaffPage(): ReactElement {
               </>
             )}
           />
-          {isUserAdmin && (
+          {canManageFacility && (
```
Comment (wittejm) [inline src/pages/StaffRosterPage.tsx:117]: shouldn't this be `canManageStaff`?

**Review by wittejm — APPROVED** [review]

File: src/pages/StaffRosterPage.tsx (hunk 4 above)
Comment (dabbler0) [reply src/pages/StaffRosterPage.tsx]: Good catch; this is also a place where I forgot to change the old "omitting" behavior to "disabled" behavior. Fixing both atonce.

File: src/pages/MessageReviewPage.tsx (hunk 2 above)
Comment (dabbler0) [reply src/pages/MessageReviewPage.tsx]: Yep, forgot about that. Changed now, although this means we can't merge this doc client until after the relevant db migration is done in the backend.

File: src/components/NotAllowed.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { LockOutlined } from "@ant-design/icons";
+import { Result } from "antd";
+import React from "react";
+
+const NotAllowed: React.FC = () => {
```
[the remark this answers is not in the document]
Comment (dabbler0) [reply src/components/NotAllowed.tsx:5]: Yep, these should only be reachable by manually typing the url.

File: src/pages/LiveCallPage/index.tsx [hunk 6]
```diff
@@ -113,181 +121,192 @@ const LiveVisitationContainer: React.FC = () => {
...
+                          className="mt-4"
+                        />
+                      ))}
+                  </div>
+                </div>
+              )}
+            </Sider>
+          </>
+        ) : (
+          <NotAllowed />
```
[the remark this answers is not in the document]
Comment (dabbler0) [reply src/pages/LiveCallPage/index.tsx]: Rearranged all the ternaries that involve `<NotAllowed />` to put `<NotAllowed />` first.

**Comment by wittejm** [comment]
🍾

**Pull request merged**